designer/server/src/test/scala/pl/touk/nussknacker/ui/validation/ProcessValidationSpec.scala

```diff
@@ -538,14 +541,139 @@ class ProcessValidationSpec extends AnyFunSuite with Matchers {
+        FlatNode(SubprocessInputDefinition("in", List(SubprocessParameter("P1", SubprocessClazzRef[Short])))),
+        FlatNode(SubprocessOutputDefinition("out", "out1", List(Field("strField", "'value'")))),
+      ),
+      additionalBranches = List.empty
+    )
+
+    val process = createProcess(
+      nodes = List(
+        Source("source", SourceRef(sourceTypeName, Nil)),
+        SubprocessInput("subIn", SubprocessRef(
+          fragmentDefinition.id,
+          List(evaluatedparam.Parameter("P1", "123"))),
+          isDisabled = Some(false)
+        ),
+        Sink("sink", SinkRef(sinkTypeName, Nil))
+      ),
+      edges = List(
+        Edge("source", "subIn", None),
+        Edge("subIn", "sink", Some(EdgeType.SubprocessOutput("out1")))
+      )
+    )
+
+    val processValidation = processValidationWithConfig(fragmentDefinition, configWithValidators)
+    val result = processValidation.validate(process)
+    result.hasErrors shouldBe false
+    result.errors.invalidNodes shouldBe Symbol("empty")
+    result.errors.globalErrors shouldBe Symbol("empty")
+    result.saveAllowed shouldBe true
+  }
+
+  test("validates scenario with fragment parameters - P1 as mandatory param with with missing actual value") {
+    val subprocessId = "fragment1"
+
+    val configWithValidators: Config = defaultConfig
+      .withValue(s"componentsUiConfig.$subprocessId.params.P1.validators", fromIterable(List(Map("type" -> "MandatoryParameterValidator").asJava).asJava))
+
+    val fragmentDefinition: CanonicalProcess = CanonicalProcess(MetaData(subprocessId, FragmentSpecificData()), List(
+      FlatNode(SubprocessInputDefinition("in", List(SubprocessParameter("P1", SubprocessClazzRef[Short])))),
+      FlatNode(SubprocessOutputDefinition("out", "out1", List(Field("strField", "'value'")))),
+    ),
+      additionalBranches = List.empty
+    )
+
+    val process = createProcess(
+      nodes = List(
+        Source("source", SourceRef(sourceTypeName, Nil)),
+        SubprocessInput("subIn", SubprocessRef(
+          fragmentDefinition.id,
+          List(evaluatedparam.Parameter("P1", ""))),
+          isDisabled = Some(false)
+        ),
+        Sink("sink", SinkRef(sinkTypeName, Nil))
+      ),
+      edges = List(
+        Edge("source", "subIn", None),
+        Edge("subIn", "sink", Some(EdgeType.SubprocessOutput("out1")))
+      )
+    )
+
+    val processValidation = processValidationWithConfig(fragmentDefinition, configWithValidators)
+    val result = processValidation.validate(process)
+
+    result.hasErrors shouldBe true
+    result.errors.globalErrors shouldBe empty
+    result.errors.invalidNodes.get("subIn") should matchPattern {
+      case Some(List(NodeValidationError("EmptyMandatoryParameter", _, _, Some("P1"), NodeValidationErrorType.SaveAllowed))) =>
+    }
+  }
+
+  test("validates scenario with fragment parameters - P1 and P2 as mandatory params with missing actual values") {
```

Comment:
maybe let's rename this test to be visible that it tests that errors during parameters validations are accumulated